Client/WebFilters/AdBlockEngineManager.swift

```diff
@@ -232,212 +122,119 @@ public actor AdBlockEngineManager: Sendable {
       version: version,
       relativeOrder: relativeOrder
     )
-    await data.add(resource: resourceWithVersion)
+    
+    add(resource: resourceWithVersion)
   }
   
   /// Tells this manager to remove all resources for the given source next time it compiles this engine
   func removeResources(for source: Source, resourceTypes: Set<ResourceType> = Set(ResourceType.allCases)) async {
-      await data.removeResources(for: source, resourceTypes: resourceTypes)
+    self.enabledResources = self.enabledResources.filter { resourceWithVersion in
+      let resource = resourceWithVersion.resource
+      guard resource.source == source && resourceTypes.contains(resource.type) else { return true }
+      return false
+    }
   }
   
   /// Start a timer that will compile resources if they change
   public func startTimer() {
     guard endlessBuildTask == nil else { return }
     
-    self.endlessBuildTask = Task.detached {
-      try await withTaskCancellationHandler(operation: {
+    self.endlessBuildTask = Task(priority: .background) {
+      do {
         while true {
```

Comment:
Use a timer instead of infinite loop. The timer will run at your interval and start a task. Otherwise you're infinitely blocking whatever core/thread this task is ran on.

Reply:
I used to do this on a detached Task. Perhaps I incorrectly removed that part. There is no reason I can put it back

Reply:
I'd rather something like:

Timer(sleepTime,repeating: true) {
    YourCodeHere()
}

This is because the timer will dispatch to a queue once triggered, rather than blocking an entire queue with a loop.

Reply:
The thing is that I don't think timers work in actors. I've never been able to get one working and the explanation I can give you is that actors don't use run loops and hence timers never trigger in an actor.

Reply:
Also I don't think Timer.sleep blocks an entire queue. The `sleep` function states in its documentation "This function doesn't block the underlying thread."